src/tests.rs

```diff
@@ -31,25 +31,36 @@ impl Node {
         Node { item, children: Vec::new() }
     }
 
-    fn into_tree(self) -> Result<Tree> {
-        fn inflate(tree: &mut Tree, parent_guid: &Guid, node: Node) -> Result<()> {
+    fn into_builder(self) -> Result<Builder> {
+        fn inflate(b: &mut Builder, parent_guid: &Guid, node: Node) -> Result<()> {
             let guid = node.item.guid.clone();
-            tree.insert(ParentGuidFrom::default()
-                            .children(&parent_guid)
-                            .item(&parent_guid),
-                        node.item.into())?;
+            match b.item(node.item) {
```

Comment:
nit: map + or_else is maybe nicer? #negligble

```suggestion
            b.item(node.item)
                .map(|_| ())
                .or_else(|err| match err.kind() {
                    ErrorKind::DuplicateItem(_) => Ok(()),
                    _ => Err(err),
                })?;
```